src/JWT/Builder/Extensions/PublicClaimsExtensions.cs

```diff
@@ -0,0 +1,77 @@
+﻿using JWT.Builder.Enums;
+using System;
+
+namespace JWT.Builder.Extensions
+{
+    public static class PublicClaimsExtensions
+    {
+        public static JwtBuilder ExpirationTime(this JwtBuilder builder, DateTime time)
+        {   
+            return builder.AddClaim(PublicClaimsNames.ExpirationTime, GetSecoundsSinceEpocheAsString(time));            
+        }
+
+        public static JwtBuilder Issuer(this JwtBuilder builder, string issuer)
+        {
+            return builder.AddClaim(PublicClaimsNames.Issuer, issuer);
+        }
+
+        public static JwtBuilder Subject(this JwtBuilder builder, string subject)
+        {
+            return builder.AddClaim(PublicClaimsNames.Subject, subject);
+        }
+
+        public static JwtBuilder Audience(this JwtBuilder builder, string audience)
+        {
+            return builder.AddClaim(PublicClaimsNames.Audience, audience);
+        }
+
+        public static JwtBuilder NotBefore(this JwtBuilder builder, DateTime time)
+        {
+            return builder.AddClaim(PublicClaimsNames.NotBefore, GetSecoundsSinceEpocheAsString(time));
+        }
+
+        public static JwtBuilder IssuedAt(this JwtBuilder builder, DateTime time)
+        {
+            return builder.AddClaim(PublicClaimsNames.IssuedAt, GetSecoundsSinceEpocheAsString(time));
+        }
+        public static JwtBuilder Id(this JwtBuilder builder, Guid id)
+        {
+            return builder.AddClaim(PublicClaimsNames.JWTID, id.ToString());
+        }
+
+        public static JwtBuilder Id(this JwtBuilder builder, long id)
+        {
+            return builder.AddClaim(PublicClaimsNames.JWTID, id.ToString());
+        }
+        public static JwtBuilder Id(this JwtBuilder builder, string id)
+        {
+            return builder.AddClaim(PublicClaimsNames.JWTID, id);
+        }
+
+        public static JwtBuilder GivenName(this JwtBuilder builder, string name)
+        {
+            return builder.AddClaim(PublicClaimsNames.GivenName, name);
+        }
+
+        public static JwtBuilder SurName(this JwtBuilder builder, string lastname)
+        {
+            return builder.AddClaim(PublicClaimsNames.SurName, lastname);
+        }
+
+        public static JwtBuilder MiddleName(this JwtBuilder builder, string middleName)
+        {
+            return builder.AddClaim(PublicClaimsNames.MiddleName, middleName);
+        }
+
+        private static double GetSecoundsSinceEpoche(DateTime time)
```

Comment:
It's a better idea to use `DateTimeOffset` everywhere, instead of `DateTime`. Official guidance from Microsoft: https://docs.microsoft.com/en-us/dotnet/standard/datetime/choosing-between-datetime

If you use `DateTimeOffset`, then there is this method: https://msdn.microsoft.com/en-us/library/system.datetimeoffset.tounixtimeseconds(v=vs.110).aspx

nitpick: `GetSecondsSinceEpoch` is the correct spelling.